src/frontend-samples/issues-sample/IssuesApi.ts

```diff
@@ -36,37 +35,25 @@ export default class IssuesApi {
     return labels.map((label, index) => ({ ...label, id: instanceKeys[index].id }));
   }
 
-  public static decoratorIsSetup() {
-    return (null != this._issuesPinDecorator);
-  }
-
   public static setupDecorator() {
-    if (undefined === this._issuesPinDecorator)
-      this._issuesPinDecorator = new MarkerPinDecorator();
+    return new MarkerPinDecorator();
```

Comment:
You can just make this a static variable that's instantly initialized. Its never set or deleted so a lot of code can be removed then.

Reply:
I think that could be a potential race condition issue for the teardown. Right now, two instances of the widget are being created for each of the samples, one of which is torn down right away. That could potentially tear down the static decorator after the "real" widget sets it up. I could add some sort of check to figure out which widget is actually being used, but I feel like that is adding more complication than just having the widget maintain the decorator.

Reply:
I'm not following, the decorator would never be deallocated or torn down if its a static variable. If `disabledecorations()` is the problem, then it would still be a problem with the current code as it would just leave the 'old' decorator in viewmanager before rendering the second widget.

Reply:
Sorry I could see how my wording was confusing above. But yes, disableDecorations() is the problem if we went with a static approach, since the widget that is torn down in the beginning will remove the static decorator from the viewManager, potentially after the other widget has attempted to add it to the viewManager. But with the current approach, both widgets are creating their own decorator, so the widget that is torn down will only remove it's own decorator from the viewManager. The widget that remains for the duration of the sample wouldn't be impacted at all. The only downside of this I could see would be that for a brief period, there could be two versions of the marker pin decorator registered with the viewManager.